Assets/Scripts/GunExtension.cs

```diff
@@ -14,12 +14,27 @@ public class GunExtension : GunModifier
     [SerializeField]
     public Transform[] outputs;
 
-    // Attaches the models to each of the attachment points of the barrel. Since a barrel can have multiple outputs (minigun)
-    public void AttachToTransforms(Transform[] transforms)
+    /// <summary>
+    /// Attaches the extension model(s) to each of the attachment points of the barrel.
+    /// Required since a barrel can have multiple outputs (minigun).
+    /// Note that the extension should be instantiated once before this method is called.
+    /// </summary>
+    /// <param name="transforms">The barrel's attachment points</param>
+    /// <returns>Combined output transforms from barrel and extension</returns>
+    public Transform[] AttachToTransforms(Transform[] transforms)
     {
-        foreach(var t in transforms[1..])
+        var attachedOutputs = new List<Transform>();
+        foreach (var t in transforms[1..])
         {
             Instantiate(model, t.position, t.rotation, transform);
+            foreach (var output in outputs)
+            {
+                // TODO Do we need a rotational offset?
+                var offset = output.position - transforms[0].position;
+                var outputInstance = Instantiate(output, t.position + offset, t.rotation, transform);
+                attachedOutputs.Add(outputInstance.transform);
```

Comment:
Resolved Sondre's complaint about extension instantiation, though I've left a TODO that is irrelevant right now. We might need to change this bit for future spicy rotations in extension outputs :)

Reply:
Made an attempt that did not work properly:

```cs
var position = t.position + (output.position - transforms[0].position);
var rotation = (output.rotation * Quaternion.Inverse(transforms[0].rotation)) * t.rotation;
var outputInstance = Instantiate(output, position, rotation, transform);
```

Reply:
Can be solved later, I would say it's ok to leave the TODO.